pkg/armhelpers/deployments.go

```diff
@@ -21,25 +21,26 @@ func (az *AzureClient) DeployTemplate(ctx context.Context, resourceGroupName, de
 
 	log.Infof("Starting ARM Deployment (%s). This will take some time...", deploymentName)
 	future, err := az.deploymentsClient.CreateOrUpdate(ctx, resourceGroupName, deploymentName, deployment)
-	if err != nil {
-		return de, err
+
+	// wait till future completes
+	e1 := future.WaitForCompletion(ctx, az.deploymentsClient.Client)
+	if err != nil && e1 != nil {
+		return resources.DeploymentExtended{}, err
```

Comment:
Is there never a case where `az.deploymentsClient.CreateOrUpdate` succeeds but an error is returned from `future.WaitForCompletion`? 

We never return the `e1` or `e2` error so I worried that might lose some information. Should we at least log the error we don't return?

Reply:
If an `err` is returned from `az.deploymentsClient.CreateOrUpdate`, is it safe to keep using the `future` instance as we do here?

Reply:
That makes sense. If we're sure calling `future.WaitForCompletion` will be safe even when `err` was returned (I tried to follow the code but couldn't quite locate the implementation in autorest-go), and if `e1` and `e2` are logged, then :+1:.